dcicutils/es_utils.py

```diff
@@ -86,3 +86,32 @@ def execute_lucene_query_on_es(client, index, query):
     except KeyError:
         logger.error('Searching index %s with query %s gave no results' % (index, query))
         return None
+
+
+def get_bulk_uuids_embedded(client, index, uuids):
+    """
+    Gets the embedded view for all uuids in an index with a single multi-get ES request.
+
+    NOTE: because an index is required, when passing uuids to this method they must all be
+    of the same item type. The index can be determined by:
+            ''.join([eb_env_name, item_type])
+            ex: fourfront-mastertestuser or fourfront-mastertestfile_format
+
+    :param client: elasticsearch client
+    :param index: index to search
+    :param uuids: list of uuids (all of the same type)
+
+    :returns: list of embedded views of the given uuids, if any
+    """
+    final_result = []
+    try:
+        response = client.mget(body={
+            'docs': [{'_id': _id,
+                      'source': ['embedded.*'],
+                      '_index': index} for _id in uuids]
+        })
+        for doc in response['docs']:
```

Comment:
Any concern about huge list?  Might set up as a generator and have a wrapper function to return a list with the option to use the generator directly if the result is huge?  similar to how search_metadata is set up.

Reply:
To @aschroed's comment...

(1) Is it possible that if it works this way, and its stack is made explicit, it could return the info retrieved in the time allotted _and_ a representation of partial state to resume from in order to continue?

(2) Is it possible that it could return json-lines format streaming and that this would get around the issue of a large list _and_ timeouts?